pkg/component/component.go

```diff
@@ -0,0 +1,85 @@
+package component
+
+import (
+	"strings"
+
+	"github.com/improbable-eng/thanos/pkg/store/storepb"
+)
+
+// Component that implements gRPC StoreApi
```

Comment:
So in Golang for public structs and methods needs to start with method or struct name (: 

So here would be `StoreAPI is an component that...` (: 

Not sure if it is written anywhere, but essentially golint (`go get -u golang.org/x/lint/golint && golint ./pkg/component/....`) will show you the guidance there. It is super strict and sometimes annoying so we don't use it in CI but for those struct and method comments it makes sense, so can we adjust comments in this file in this way? (:

Reply:
And missing trailing period for all comments here (: